### ISSUE TITLE
Refactor `ops` to use `async_session` instead of `sync_session`

### ISSUE THREAD

**Issue by ThomasLaPiana** [ethyca/fides#3092]
### Is your feature request related to a specific problem?

Currently, the `ctl`-related code uses async database sessions while the `ops`-related code uses sync sessions. Apart from just the performance considerations here, this is an annoying bit of tech debt that hurts testing/interoperability/understandability

This should be refactor so that everything uses `async_session`

**Comment by pattisdr** [comment]
My two cents, if we're converting ops-side to use async sessions instead of ctl side to use sync sessions, I think that this is going to be a lot larger effort than a 5, particularly given how we test things ops-side.

**Comment by ThomasLaPiana** [comment]
@pattisdr Agreed, I know it's going to be a huge lift! I'll update the points

I'm still insistent to use `async` though across the application as that is "the right way" in my opinion. Otherwise I foresee a day where we move everything from `sync` back over to `async` again due to performance issues or something else.
